cou/steps/ceph.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2024 Canonical Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Functions for prereq steps related to ceph."""
+import json
+import logging
+
+from cou.exceptions import ApplicationError, ApplicationNotFound, UnitNotFound
+from cou.utils.juju_utils import Model
+
+logger = logging.getLogger(__name__)
+
+
+async def _get_leader_unit_name(model: Model) -> str:
```

Comment:
That's a really good point about multiple ceph applications - I think this might be possible in regular deployments(?).  So perhaps we need to iterate over all the ceph-mon applications found?

Reply:
okay, let me change to any unit (which also save me from adding `leader` field in `Unit` class)

Reply:
Migrated